### PR TITLE
Add google places

### PR DESCRIPTION
**Pull request by Kalior**
Add places with google place id

### CODE REVIEW

File: src/components/PlaceView.tsx [hunk 1]
```diff
@@ -49,24 +41,38 @@ export const PlaceView = ({
...
+        <StarRating rating={rating} />
+      </StarCell>
+      <RollReviews reviews={reviews} />
+    </PlaceBannerContent>
+  );
+
+  const banner = place.googlePlace ? (
+    <PlaceBanner
+      key={place.googlePlace.photos[0].photo_reference}
+      url={`https://maps.googleapis.com/maps/api/place/photo?maxwidth=1200&photoreference=${
```
Comment (lindskogen) [inline src/components/PlaceView.tsx]: We should put all google place api related code and strings in another file

File: src/components/PlaceView.tsx [hunk 2]
```diff
@@ -138,8 +144,16 @@ const PlaceBanner = styled.div`
...
   height: 15em;
 
-  background-color: #fff;
   box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.4);
+
+  text-shadow: -1px 0 2px white, 0 1px 2px white, 1px 0 2px white,
+    0 -1px 2px white;
+  font-weight: 500;
+
+  background: ${props => (props.url ? `url(${props.url})` : "#fff")};
```
Comment (lindskogen) [inline src/components/PlaceView.tsx]: Nice!

File: src/components/StatsController.tsx [hunk 3]
```diff
@@ -80,12 +80,18 @@ export const StatsController = ({ userId }: Props) => {
   };
 
   const handleNewPlaceInput = (event: ChangeEvent<HTMLInputElement>) => {
-    if (!event.target.name) {
+    if (!event.target || !event.target.name) {
       throw new Error("No name on event.target");
     }
+    if (event.persist) {
+      event.persist();
```
Comment (lindskogen) [inline src/components/StatsController.tsx]: oh no...

Går det att lösa utan?

Comment (lindskogen) [reply src/components/StatsController.tsx]: Spara ner target.name och target.value i variabler innan

File: src/components/StatsController.tsx [hunk 4]
```diff
@@ -80,12 +80,18 @@ export const StatsController = ({ userId }: Props) => {
...
+    }
     console.log(event.target.name);
     console.log(event.target.value);
```
Comment (lindskogen) [inline src/components/StatsController.tsx]: Remove console.log :)

File: src/lib/backend.ts [hunk 5]
```diff
@@ -28,8 +28,9 @@ export const getPlaceById = (placeId: string): Promise<Place> => {
 };
 
 export const postPlace = (place: Partial<Place>): Promise<Place> => {
-  const { placeName, google_maps_link, comment } = place;
-  if (!placeName || !google_maps_link || !comment) {
+  console.log("Posting", place);
```
Comment (lindskogen) [inline src/lib/backend.ts]: Remove console.log :)

File: serverless-api/src/repository/places.ts [hunk 6]
```diff
@@ -41,6 +41,24 @@ export const getPlaceByName = async (
...
+): Promise<Place | undefined> => {
+  const queryParams = {
+    TableName: "Places",
+    IndexName: "placeGoogleIdIndex",
+    KeyConditionExpression: "googlePlaceId = :google_place_id",
+    ExpressionAttributeValues: { ":google_place_id": googlePlaceId }
+  };
+  const res = await dynamodb.query(queryParams).promise();
+
+  if (!res.Items || !res.Items[0]) {
```
Comment (lindskogen) [inline serverless-api/src/repository/places.ts]:
```suggestion
  if (!res.Items) {
```

Undefined check in 2 places

File: serverless-api/src/ratings.ts [hunk 7]
```diff
@@ -16,20 +18,37 @@ export const getPlacesWithRatings: LambdaHandler = async () => {
...
+        const sum = reviews.reduce((sum, review) => review.rating + sum, 0);
+
+        const rating = {
+          ...place,
+          rating: sum / reviews.length
+        };
+
+        if (place.googlePlaceId) {
+          const googlePlace = await fetch(
+            `https://maps.googleapis.com/maps/api/place/details/json?placeid=${
```
Comment (lindskogen) [inline serverless-api/src/ratings.ts]: We should put all google place api related code and strings in another file

File: serverless-api/src/places.ts [hunk 8]
```diff
@@ -39,27 +42,63 @@ export const getById: LambdaHandler = async (event, context) => {
...
+
+  if (googlePlaceId) {
+    const place = await getPlaceByGoogleId(googlePlaceId);
+    if (place) {
+      return createResponse(409, {
+        error: "Place with googlePlaceId already exists"
+      });
+    } else {
+      const googlePlace = await fetch(
+        `https://maps.googleapis.com/maps/api/place/details/json?placeid=${googlePlaceId}&key=${
```
Comment (lindskogen) [inline serverless-api/src/places.ts]: We should put all google place api related code and strings in another file

File: serverless-api/src/places.ts [hunk 9]
```diff
@@ -39,27 +42,63 @@ export const getById: LambdaHandler = async (event, context) => {
   if (!place) {
     return createResponse(404, { message: "No place with that id" });
   }
+
+  if (place.googlePlaceId) {
+    const googlePlace = await fetch(
+      `https://maps.googleapis.com/maps/api/place/details/json?placeid=${
```
Comment (lindskogen) [inline serverless-api/src/places.ts]: We should put all google place api related code and strings in another file

File: src/lib/backend.ts (hunk 5 above)
Comment (Kalior) [inline src/lib/backend.ts]:
```suggestion

```

**Pull request merged**